Fix Key Vault project files

heaths opened the pull request.

heaths committed:
Fix Key Vault project files

heaths commented:
/azp run net - core - ci

heaths committed:
Remove test strategy breaking build

The Windows_NetCoreApp_ProjectReferences test strategy is breaking the
build because it's converting PackageReferences to ProjectReferences,
and for recorded tests this is failing the build beacuse Secrets is now
a preview version, which then causes Azure.Identity to fail because
records are wrong.

chidozieononiwu commented:
We probably just need to update some recorded tests in Azure.Identity.
 ServiceVersion in CryptographyClientOptions.cs https://github.com/Azure/azure-sdk-for-net/blob/cddefe7d93026be8b4ad835d0cb7c64969c4ff46/sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClientOptions.cs#L19 has moved ahead of what is present in our recorded tests.
https://github.com/Azure/azure-sdk-for-net/blob/b015ef33bf4945661d74c4f18411e986558fbefe/sdk/identity/Azure.Identity/tests/SessionRecords/ManagedIdentityCredentialImdsLiveTests/ValidateImdsUserAssignedIdentity.json#L4

This should unblock for now although I don't think reverting the KeyVault package versions make a difference

heaths commented:
Then all the tests using PackageReferences would break, which are 4 platforms. I think the long term solution needs more thought to how this could work when service versions rev.

weshaggard commented on eng/pipelines/templates/jobs/archetype-sdk-tests.yml:
```diff
@@ -24,11 +24,11 @@ jobs:
-          OSVmImage: "windows-2019"
-          TestTargetFramework: netcoreapp2.1
-          ConvertToProjectReferenceOption: /p:UseProjectReferenceToAzureClients=true
+        # Windows_NetCoreApp_ProjectReferences:
```
We should definitely not comment this out. As we chatted in IM we should update the test projects.

weshaggard commented on sdk/keyvault/Azure.Security.KeyVault.Keys/src/Azure.Security.KeyVault.Keys.csproj:
```diff
@@ -3,7 +3,6 @@
   <PropertyGroup>
     <Description>This is the Microsoft Azure Key Vault Keys client library</Description>
     <AssemblyTitle>Microsoft Azure.Security.KeyVault.Keys client library</AssemblyTitle>
-    <Version>4.0.2</Version>
```
Any idea how this bad merge happened? Was it incorrectly merged? I would have expected a merge conflict so we would have needed to take both in order to get into this state.

heaths replied:
Not sure how it got merged that way, but the effect would've been the same since the next `Version` would've overwrote this first one. Could have something to do with .gitattributes using `merge=union` for *.csproj. In most other repos, .csproj are treated as any other text file so a merge conflict would've happened (if not resolved automatically). Can't say I've seen this before without `merge=union`.

weshaggard replied:
Thanks for the pointer I think you are correct that is what caused this. After some reading of what union does I don't think we want that for our repo so I'm going to remove it.

weshaggard replied:
https://github.com/Azure/azure-sdk-for-net/pull/10440

heaths committed:
Hard code SecretClient to use service version 7.0

Otherwise, converting PackageReferences to ProjectReferences causes recorded tests to fail using the latest SecretClient version from a preview we just shipped.

This also reverts commit c9fa0a01363ffc55eef39e3d6c3430bb8e888dd7.
